client/my-sites/checkout/modal/index.tsx

```diff
@@ -0,0 +1,101 @@
+import { StripeHookProvider } from '@automattic/calypso-stripe';
+import { Modal } from '@wordpress/components';
+import { removeQueryArgs } from '@wordpress/url';
+import { useTranslate } from 'i18n-calypso';
+import page from 'page';
+import { useSelector } from 'react-redux';
+import CheckoutMasterbar from 'calypso/layout/masterbar/checkout';
+import { getStripeConfiguration } from 'calypso/lib/store-transactions';
+import CalypsoShoppingCartProvider from 'calypso/my-sites/checkout/calypso-shopping-cart-provider';
+import CompositeCheckout from 'calypso/my-sites/checkout/composite-checkout/composite-checkout';
+import getPreviousRoute from 'calypso/state/selectors/get-previous-route.js';
+import isAtomicSite from 'calypso/state/selectors/is-site-automated-transfer';
+import { isJetpackSite } from 'calypso/state/sites/selectors';
+import { getSelectedSite, getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { KEY_PRODUCTS } from './constants';
+import type { FunctionComponent } from 'react';
+
+import './style.scss';
+
+export interface Props {
```

Comment:
This has a bunch of props but we only ever pass one here (and even it's null):

https://github.com/Automattic/wp-calypso/blob/87be0f970d1659904d633cb57144e4d8e8e4f31f/client/signup/steps/design-picker/index.jsx#L172

, am I missing something here? 😄

Reply:
I think it's fine as all of the props are optional and we can even remove passing `checkoutOnSuccessCallback`.

Reply:
No my question was more like, why did we even introduce props in a PR which are never used? (Why not introduce later when they are actually required)

Reply:
I think this is an accident and I'll remove it